src/vision.cs

```diff
@@ -0,0 +1,792 @@
+
+		// TODO: Enable once CoreImage Xcode 9 is bound
+		//[NullAllowed, Export ("barcodeDescriptor", ArgumentSemantic.Strong)]
+		//CIBarcodeDescriptor BarcodeDescriptor { get; }
+	}
+
+	[TV (11,0), Mac (10,13, onlyOn64: true), iOS (11,0)]
+	[BaseType (typeof (VNObservation))]
+	interface VNHorizonObservation {
+
+		[Export ("transform", ArgumentSemantic.Assign)]
+		CGAffineTransform Transform { get; }
+
+		[Export ("angle")]
+		nfloat Angle { get; }
+	}
+
+	[TV (11,0), Mac (10,13, onlyOn64: true), iOS (11,0)]
+	[Abstract]
+	[BaseType (typeof (VNObservation))]
+	interface VNImageAlignmentObservation {
+	}
+
+	[TV (11,0), Mac (10,13, onlyOn64: true), iOS (11,0)]
+	[BaseType (typeof (VNImageAlignmentObservation))]
+	interface VNImageTranslationAlignmentObservation {
+
+		[Export ("alignmentTransform", ArgumentSemantic.Assign)]
+		CGAffineTransform AlignmentTransform { get; set; }
+	}
+
+	[TV (11,0), Mac (10,13, onlyOn64: true), iOS (11,0)]
+	[BaseType (typeof (VNImageAlignmentObservation))]
+	interface VNImageHomographicAlignmentObservation {
+
+		[Export ("warpTransform", ArgumentSemantic.Assign)]
+		Matrix3 WarpTransform {
+			[MarshalDirective (NativePrefix = "xamarin_simd__", Library = "__Internal")]
+			get;
+			[MarshalDirective (NativePrefix = "xamarin_simd__", Library = "__Internal")]
+			set;
+		}
+	}
+
+	[TV (11,0), Mac (10,13, onlyOn64: true), iOS (11,0)]
+	[Abstract]
+	[BaseType (typeof (NSObject))]
+	interface VNRequest : NSCopying {
+
+		[Export ("initWithCompletionHandler:")]
+		[DesignatedInitializer]
+		IntPtr Constructor ([NullAllowed] VNRequestCompletionHandler completionHandler);
+
+		[Export ("preferBackgroundProcessing")]
+		bool PreferBackgroundProcessing { get; set; }
+
+		[NullAllowed, Export ("preferredMetalContext", ArgumentSemantic.Retain)]
+		IMTLDevice PreferredMetalContext { get; set; }
+
+		[NullAllowed, Export ("results", ArgumentSemantic.Copy)]
+		VNObservation [] Results { get; }
+
+		[NullAllowed, Export ("completionHandler", ArgumentSemantic.Copy)]
+		VNRequestCompletionHandler CompletionHandler { get; }
+	}
+
+	[TV (11,0), Mac (10,13, onlyOn64: true), iOS (11,0)]
+	[Abstract]
+	[BaseType (typeof (VNRequest))]
+	interface VNImageBasedRequest {
+
+		[Export ("initWithCompletionHandler:")]
+		[DesignatedInitializer]
+		IntPtr Constructor ([NullAllowed] VNRequestCompletionHandler completionHandler);
+
+		[Export ("regionOfInterest", ArgumentSemantic.Assign)]
+		CGRect RegionOfInterest { get; set; }
+	}
+
+	[TV (11,0), Mac (10,13, onlyOn64: true), iOS (11,0)]
+	enum VNImageOption {
+		[Field ("VNImageOptionProperties")]
+		Properties,
+
+		[Field ("VNImageOptionCameraIntrinsics")]
+		CameraIntrinsics,
+	}
+
+	[TV (11,0), Mac (10,13, onlyOn64: true), iOS (11,0)]
+	[BaseType (typeof (NSObject))]
+	[DisableDefaultCtor]
+	interface VNImageRequestHandler {
+
+		[Export ("initWithCVPixelBuffer:options:")]
+		IntPtr Constructor (CVPixelBuffer pixelBuffer, NSDictionary<NSString, NSObject> options);
```

Comment:
use strong dictionary
which means the generic version of NSDictionary might not be the best choice